test/odbc/psqlodbc/bigint.cpp

```diff
@@ -0,0 +1,411 @@
+
+  for (int i = 1; i < NUM_OF_DATA; i++) {
+    curr = StringToBigInt(INSERTED_DATA[i]);
+    sum += curr;
+
+    min_expected = std::min(curr, min_expected);
+    max_expected = std::max(curr, max_expected);
+  }
+  expected_results.push_back(min_expected);
+  expected_results.push_back(max_expected);
+  expected_results.push_back(sum);
+  expected_results.push_back(sum / NUM_OF_DATA);
+
+  // Create a vector of length NUM_OF_OPERATIONS with dummy value of -1 to store column results
+  vector<long long int> col_results(NUM_OF_OPERATIONS, -1);
+  
+  createTable(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS);
+  insertValuesInTable(ServerType::PSQL, TABLE_NAME, insertString, NUM_OF_DATA);
+  testComparisonFunctions(ServerType::PSQL, TABLE_NAME, SQL_C_SBIGINT, col_results, bufferLen, OPERATIONS_QUERY, expected_results, expectedLen);
+  dropObject(ServerType::PSQL, "TABLE", TABLE_NAME);
+}
+
+TEST_F(PSQL_DataTypes_BigInt, View_Creation) {
+  const string VIEW_QUERY = "SELECT * FROM " + TABLE_NAME;
+
+  long long int data;
+  const int bufferLen = 0;
+
+  const vector<string> INSERTED_DATA = {
+    "9223372036854775807",
+    "123456789",
+    "-9223372036854775808"
+  };
+  
+  const vector<long long int> EXPECTED_DATA = {
+    StringToBigInt("9223372036854775807"),
+    StringToBigInt("123456789"),
+    StringToBigInt("-9223372036854775808")
+  };
+
+  const vector<long> expectedLen(EXPECTED_DATA.size(), BIGINT_BYTES_EXPECTED);
+
+  createTable(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS);
+  testInsertionSuccess(ServerType::PSQL, TABLE_NAME, COL1_NAME, SQL_C_SBIGINT, data, bufferLen, INSERTED_DATA, EXPECTED_DATA, expectedLen);
+
+  createView(ServerType::PSQL, VIEW_NAME, VIEW_QUERY);
+  verifyValuesInObject(ServerType::PSQL, VIEW_NAME, COL1_NAME, SQL_C_SBIGINT, data, bufferLen, INSERTED_DATA, EXPECTED_DATA, expectedLen);
+  
+  dropObject(ServerType::PSQL, "VIEW", VIEW_NAME);
+  dropObject(ServerType::PSQL, "TABLE", TABLE_NAME);
+}
+
+TEST_F(PSQL_DataTypes_BigInt, Table_Unique_Constraints) {
+  const vector<pair<string, string>> TABLE_COLUMNS = {
+    {COL1_NAME, "INT PRIMARY KEY"},
+    {COL2_NAME, DATATYPE_NAME + " UNIQUE"}
+  };
+
+  const vector<string> UNIQUE_COLUMNS = {COL2_NAME};
+
+  string tableConstraints = createTableConstraint("UNIQUE", UNIQUE_COLUMNS);
+
+  long long int data;
+  const int bufferLen = 0;
+
+  const vector<string> INSERTED_DATA = {
+    "9223372036854775807",
+    "123456789"
+  };
+  const vector<long long int> EXPECTED_DATA = {
+    StringToBigInt("9223372036854775807"),
+    StringToBigInt("123456789")
+  };
+
+  const vector<long> expectedLen(EXPECTED_DATA.size(), BIGINT_BYTES_EXPECTED);
+
+  createTable(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS, tableConstraints);
+  testUniqueConstraint(ServerType::PSQL, TABLE_NAME, UNIQUE_COLUMNS);
+  testInsertionSuccess(ServerType::PSQL, TABLE_NAME, COL1_NAME, SQL_C_SBIGINT, data, bufferLen, INSERTED_DATA, 
```

Comment:
Tests should try to insert values that do NOT work to test the unique constraint
1. Insert some values (Should succeed)
2. Insert the same values (Should not succeed)

Reply:
Done.